src/logger-compat.ts

```diff
@@ -20,23 +20,43 @@
 
 import * as is from 'is';
 
-import {kFormat, Logger, LoggerConfig} from './logger';
+import {kFormat, LEVELS, Logger, LoggerConfig} from './logger';
+
+export interface CustomLevelsLoggerConfig extends LoggerConfig {
+  /**
+   * The list of levels to use.
+   */
+  levels: string[];
+}
+
+// tslint:disable:no-any
+export type CustomLevelsLoggerResult = {
+  [logLevel: string]: (...args: any[]) => CustomLevelsLoggerResult;
+}&{
+  format: (...args: any[]) => string;
+  levels: string[];
+  level: string|false;
+};
+// tslint:enable:no-any
 
 // tslint:disable-next-line:no-any
 function isString(obj: any): obj is string {
   return is.string(obj);
 }
 
-function createLogger(optionsOrLevel?: Partial<LoggerConfig>|string) {
+function createLogger(optionsOrLevel?: Partial<CustomLevelsLoggerConfig>|
+                      string): CustomLevelsLoggerResult {
   // Canonicalize input.
   if (isString(optionsOrLevel)) {
     optionsOrLevel = {
       level: optionsOrLevel,
     };
   }
-  const options: LoggerConfig =
-      Object.assign({}, Logger.DEFAULT_OPTIONS, optionsOrLevel);
-  const result = new Logger(options);
+  const options: CustomLevelsLoggerConfig =
+      Object.assign({levels: LEVELS}, Logger.DEFAULT_OPTIONS, optionsOrLevel);
+  // ts: We construct other fields on result after its declaration.
+  // tslint:disable-next-line:no-any
+  const result: CustomLevelsLoggerResult = new Logger(options) as any;
```

Comment:
I think you can use `{}` instead of `any` here.

Reply:
I tried it and it doesn't work, probably because `{}` cannot be casted to `CustomLevelsLoggerResult`.